brainiak/hyperparamopt/hpo.py

```diff
@@ -0,0 +1,231 @@
+
+    maxlimit : double, default : +inf
+      Maximum limit for the distribution
+
+    weights : double scalar or 1D array with same size as x, default 1.0
+      Used to weight the points non-uniformly if required
+    """
+
+    def __init__(self, x, minlimit=-np.inf, maxlimit=np.inf, weights=1.0):
+        self.points = x
+        self.N = x.size
+        self.minlimit = minlimit
+        self.maxlimit = maxlimit
+        self.sigma = getsigma(x, minlimit=minlimit, maxlimit=maxlimit)
+        self.weights = 2. / (erf((maxlimit - x)
+                             / (np.sqrt(2.) * self.sigma))
+                             - erf((minlimit - x)
+                             / (np.sqrt(2.) * self.sigma))) * weights
+        # return self
+
+    def __call__(self, xt):
+        if (np.isscalar(xt)):
+            return getgmmpdf(xt, self.points, self.sigma, self.weights,
+                             self.minlimit, self.maxlimit)
+        else:
+            return np.array([getgmmpdf(t, self.points, self.sigma,
+                                       self.weights, self.minlimit,
+                                       self.maxlimit) for t in xt])
+
+    def get_samples(self, chains=1, points_per_chain=1, burn_in=2000):
+        pts = get_multichain_samples(N=points_per_chain,
+                                     p=self, nchains=chains, burn_in=burn_in)
+        return pts
+
+
+def getNextSample(x, y, minlimit=-np.inf, maxlimit=np.inf):
+    z = np.array(list(zip(x, y)), dtype=np.dtype([('x', float), ('y', float)]))
+    z = np.sort(z, order='y')
+    n = y.shape[0]
+    g = int(np.round(np.ceil(0.15 * n)))
+    ldata = z[0:g]
+    gdata = z[g:n]
+    lymin = ldata['y'].min()
+    lymax = ldata['y'].max()
+    weights = (lymax - ldata['y']) / (lymax - lymin)
+    lx = gmm_1d_distribution(ldata['x'], minlimit=minlimit,
+                             maxlimit=maxlimit, weights=weights)
+    gx = gmm_1d_distribution(gdata['x'], minlimit=minlimit, maxlimit=maxlimit)
+
+    samples = lx.get_samples(chains=10, points_per_chain=100)
+    ei = lx(samples) / gx(samples)
+
+    h = (x.max() - x.min()) / (10 * x.size)
+    # assumes prior of x is uniform -- should change for different priors
+    # d = np.abs(x - samples[ei.argmax()]).min()
+    # CDF(x+d/2) - CDF(x-d/2) < 1/(10*x.size) then reject else accept
+    s = 0
+    while (np.abs(x - samples[ei.argmax()]).min() < h):
+        ei[ei.argmax()] = 0
+        s = s + 1
+        if (s == samples.size):
+            break
+    xnext = samples[ei.argmax()]
+
+    return xnext
+
+
+def getSample(x, y, dist, minlimit=-np.inf, maxlimit=np.inf):
+    if (dist == 'GMM'):
+        return getNextSample(x, y, minlimit, maxlimit)
+    if (dist == 'uniform'):
+        return np.random.random() * (maxlimit - minlimit) + minlimit
+    if (dist == 'loguniform'):
+        return np.exp(np.random.random()
+                      * (np.log(maxlimit) - np.log(minlimit))
+                      + np.log(minlimit))
+
```

Comment:
Looks like there are only supported values for dist. Should validate dist is in that list of supported values and throw an exception otherwise.